src/potions.ts

```diff
@@ -460,18 +482,64 @@ const rufusPotion = new Potion($item`closed-circuit pay phone`, {
   },
 });
 
-export const wishPotions = Effect.all()
-  .filter((effect) => !effect.attributes.includes("nohookah"))
+export const wishPotions = validWishes.map(
+  (effect) =>
+    new Potion($item`pocket wish`, {
+      effect,
+      canDouble: false,
+      duration: 20,
+      use: (quantity: number) => {
+        const s = getValidWishString(effect);
+        if (!s) return false;
+
+        for (let i = 0; i < quantity; i++) {
+          const madeValidWish = cliExecute(`genie effect ${s}`);
+          if (!madeValidWish) return false;
+        }
+        return true;
+      },
+    })
+);
+
+export const pawPotions = validWishes
   .map(
     (effect) =>
-      new Potion($item`pocket wish`, {
+      new Potion($item`cursed monkey's paw`, {
         effect,
         canDouble: false,
-        duration: 20,
-        use: (quantity: number) =>
-          new Array(quantity).fill(0).every(() => cliExecute(`genie effect ${effect}`)),
+        price: () =>
+          get("_monkeyPawWishesUsed", 0) >= 5 ||
+          !have($item`cursed monkey's paw`) ||
+          failedWishes.includes(effect)
+            ? 2 ** 100 // Something large but non-infinite for sorting reasons
+            : 0,
+        duration: 30,
+        acquire: () => (get("_monkeyPawWishesUsed", 0) >= 5 ? 0 : 1),
+        use: () => {
+          if (
+            get("_monkeyPawWishesUsed", 0) >= 5 ||
+            !have($item`cursed monkey's paw`) ||
+            failedWishes.includes(effect)
+          ) {
+            return false;
+          }
+
+          const s = getValidWishString(effect);
+          if (!s) return false;
+
+          const currentEffectTurns = haveEffect(effect);
+          visitUrl("main.php?action=cmonk&pwd");
+          runChoice(1, `wish=${s}`);
+          visitUrl("main.php");
+          if (haveEffect(effect) <= currentEffectTurns) {
```

Comment:
imo it would be better practice to regexp the result of runChoice, but that can wait for libram